scripts/Diffraction/isis_powder/polaris.py

```diff
@@ -72,13 +75,21 @@ def create_total_scattering_pdf(self, **kwargs):
         return pdf_output
 
     def set_sample_details(self, **kwargs):
+        self._set_environment_details("sample", **kwargs)
+
+    def set_container_details(self, **kwargs):
+        self._set_environment_details("container", **kwargs)
+
+    def _set_environment_details(self, environment_type: str = "sample", **kwargs):
```

Comment:
Based on the way you've set this up I wonder if it makes sense to rename the sample_details class to something general like shape_details and to use it for holding a shape and a material regardless of whether the object is the sample or a container.
I appreciate this is a rather different suggestion to what I was talking about earlier where I was wondering about having some kind of super class that could hold the details of both a sample and a container. I guess that was motivated partly by the idea of having some validation logic that checked that the sample and container shapes are consistent. But that probably ought to be checked in the PPMonteCarloAbsorption algorithm rather than you replicating that validation logic here.